pkg/util/cluster/cluster.go

```diff
@@ -258,6 +266,25 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 	return nil
 }
 
+func (c *Cluster) generateSubnets() (vnetPrefix string, masterSubnet string, workerSubnet string) {
+	f1 := 128 + rand.Intn(16000)
+	vnetPrefix, masterSubnet, workerSubnet = c.getSubnets(f1)
+	return
+}
+
+func (c *Cluster) getSubnets(num int) (vnetPrefix string, masterSubnet string, workerSubnet string) {
+	v := uint(10)<<24 + uint(num*2)<<8
+	v3 := byte(v & 0xFF)
+	v2 := byte((v >> 8) & 0xFF)
+	v1 := byte((v >> 16) & 0xFF)
+	v0 := byte((v >> 24) & 0xFF)
+
+	vnetPrefix = fmt.Sprintf("%s/23", net.IPv4(v0, v1, v2, v3))
+	masterSubnet = fmt.Sprintf("%s/24", net.IPv4(v0, v1, v2, v3))
+	workerSubnet = fmt.Sprintf("%s/24", net.IPv4(v0, v1, v2+1, v3))
+	return
+}
```

Comment:
is something like this less terrifying?
```golang
// pick a random /23 in the range [10.0.2.0, 10.128.0.0).  10.0.0.0 is used
// by dev-vnet to host CI; 10.128.0.0+ is used for pods.

var x, y int
for x == 0 && y == 0 {
	x, y = rand.Intn(128), 2*rand.Intn(128)
}

vnetPrefix = fmt.Sprintf("10.%d.%d.0/23", x, y)
masterSubnet = fmt.Sprintf("10.%d.%d.0/24", x, y)
workerSubnet = fmt.Sprintf("10.%d.%d.0/24", x, y+1)
```

Reply:
Yeah, I went with this in the first place -- I guess it is much nicer, I was just hoping that there was a nicer option than string formatting, lol